### PR TITLE
add mage to run the build/test jobs and setup initial gh actions

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- fix go module name
- add initial goreleaser (will add the sign and docker files)
- setup mage to run the build/test commands
- setup golangci-lint

#### Ticket Link

Fixes

#### Release Note

```release-note
add mage to run the build/test jobs and setup initial gh actions 
```

### CODE REVIEW

**Comment by cpanato** [comment]
will need some follow ups for sure but maybe it is a good start

/cc @developer-guy @dlorenc

**Comment by developer-guy** [comment]
I loved it, thank you so much for this effort, looks amazing! Every time I learn new stuff from you, new strategies 🙋🏻‍♂️

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
```
Comment (developer-guy) [inline .github/workflows/build.yml:6]: I think we should move from `master` to `main` in order to trigger this.

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: golangci-lint
+
+on:
+  push:
+    branches:
+      - main
```
Comment (developer-guy) [inline .github/workflows/golangci-lint.yml:6]: same 😋

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
```
Comment (developer-guy) [inline .github/workflows/test.yml:6]: same 😋

**Review by developer-guy — CHANGES_REQUESTED** [review]

**Comment by developer-guy** [comment]
just found minor things, once we merge this, I want to work with container image build/push stuff 🤩

File: .github/workflows/build.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/build.yml:6]: I've requested to @dlorenc to rename the default branch to `main` :)

Comment (cpanato) [reply .github/workflows/build.yml:6]: ok branch renamed :) thanks Dan

**Review by developer-guy — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -1,18 +1,19 @@
-module github.com/developer-guy/cosign-gatekeeper-provider
+module github.com/sigstore/cosign-gatekeeper-provider
 
 go 1.17
 
 require (
+	get.porter.sh/porter v0.38.8
```
Comment (dirien) [inline go.mod:6]: Are you sure to get the port lib in for the helper function in mage?

Comment (cpanato) [reply go.mod:6]: not 100% maybe i will drop that and reimplement the git variables locally, I was re-thinking on that and maybe this brings a lot of things that we will not use.
thanks for the heads up

Comment (cpanato) [reply go.mod]: i've dropped that for now and when we implement the version or other things that need ldflags I will reimplement the feature.

Comment (cpanato) [reply go.mod]: PTAL @dirien

Comment (dirien) [reply go.mod]: cool!

**Review by dirien — APPROVED** [review]
LGTM

**Pull request merged**